### PR TITLE
QMAPS-1074: Migrate Service Panel in React

### PR DESCRIPTION
**Pull request by xem**
## Description
Replace the current service panel with a React equivalent

## Why
It is part of our migration plan

## Screenshots
image

### CODE REVIEW

File: src/panel/servicePanel.jsx [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+      minimizedTitle={_('Show Qwant Maps services', 'service panel')}
+      className={'service_panel'}
+    >
+      <div className="service_panel__categories">
+        {
+          CategoryService.getCategories().map(item =>
+            <button className="service_panel__category" type="button" key={item.name}
+              onClick={() => {
+                window.app.navigateTo(`/places/?type=${item.name}`);
+              }}>
```
Comment (bbecquet) [inline src/panel/servicePanel.jsx]: When https://github.com/QwantResearch/erdapfel/pull/404 is merged you'll be able to write these callbacks inline if you prefer.

File: src/panel/servicePanel.jsx [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+import CategoryService from '../adapters/category_service';
+import nconf from '@qwant/nconf-getter';
+
+class ServicePanel extends React.Component {
+  render() {
+    return <Panel
+      resizable
+      title={_('Qwant Maps services', 'service panel')}
+      minimizedTitle={_('Show Qwant Maps services', 'service panel')}
+      className={'service_panel'}
```
Comment (bbecquet) [inline src/panel/servicePanel.jsx]: The `{ }` are not necessary. There are ESLint rules specific to JSX to help for this kind of stuff, maybe it's time to introduce them :)

File: src/panel/servicePanel.jsx [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    >
+      <div className="service_panel__categories">
+        {
```
Comment (bbecquet) [inline src/panel/servicePanel.jsx]: This part could be extracted in a separate component so it's ready to be used by the POI panel. But let's wait for it to be merged first ;)

File: src/panel/app_panel.js [hunk 4]
```diff
@@ -4,7 +4,7 @@ import PanelsView from '../views/app_panel.dot';
 import Panel from '../libs/panel';
 import FavoritePanel from './favorites_panel';
 import PoiPanel from './poi_panel';
-import ServicePanel from './service_panel';
+import ServicePanel from './servicePanel';
```
Comment (bbecquet) [inline src/panel/app_panel.js]: Just a naming convention of React. It's better to use capitalized named for component files. It's a consistency reason, because when used in JSX, custom components *must* start with a capital (lowercase is reserved for HTML tags).

File: src/panel/servicePanel.jsx (hunk 1 above)
Comment (bbecquet) [reply src/panel/servicePanel.jsx]: Done! If you prefer to inline this king of call (there are 3) for readability, you can :)

File: tests/integration/tests/menu.js [hunk 5]
```diff
@@ -61,14 +61,15 @@ test('menu open favorite', async () => {
...
+test('close service panel when clicking direction', async () => {
   expect.assertions(2);
+  await page.goto(APP_URL);
   await page.click('.service_panel__item__direction');
   const servicePanelClose = await page.waitForSelector('.service_panel', { hidden: true });
   expect(servicePanelClose).not.toBeFalsy();
 });
 
-test('service panel open on load', async () => {
+test('close service panel routes panel is opened on load', async () => {
```
Comment (bbecquet) [inline tests/integration/tests/menu.js]: Je comprends pas ce test (ni sa description).
Là on teste exactement la même chose que le précédent. Et au final dans la suite on ne teste même plus que le panel s'affiche quand on arrive sur l'app.

File: tests/integration/tests/menu.js [hunk 6]
```diff
@@ -61,14 +61,15 @@ test('menu open favorite', async () => {
   expect(favorites).not.toBeNull();
 });
 
-test('one panel open at a time', async () => {
+test('close service panel when clicking direction', async () => {
   expect.assertions(2);
+  await page.goto(APP_URL);
   await page.click('.service_panel__item__direction');
   const servicePanelClose = await page.waitForSelector('.service_panel', { hidden: true });
   expect(servicePanelClose).not.toBeFalsy();
```
Comment (bbecquet) [inline tests/integration/tests/menu.js]: Si tu regardes la doc de Puppeteer pour `waitForSelector` https://github.com/GoogleChrome/puppeteer/blob/master/docs/api.md#pagewaitforselectorselector-options, tu verras que le résultat attendu quand tu utilises `hidden: true` est `null`. Du coup il faut remplacer `not.toBeFalsy` par `toBeNull`.

File: tests/integration/tests/menu.js [hunk 7]
```diff
@@ -61,14 +61,15 @@ test('menu open favorite', async () => {
   expect(favorites).not.toBeNull();
 });
 
-test('one panel open at a time', async () => {
+test('close service panel when clicking direction', async () => {
   expect.assertions(2);
```
Comment (bbecquet) [inline tests/integration/tests/menu.js]: Il n'y a plus qu'une seule assertion dans ce test, donc si tu laisses 2 ça cassera.

File: package.json [hunk 8]
```diff
@@ -87,7 +87,7 @@
     "@qwant/merge-po": "file:local_modules/merge-po",
     "@qwant/nconf-builder": "file:local_modules/nconf_builder",
     "@qwant/po-js": "file:local_modules/po-js",
-    "@qwant/qwant-basic-gl-style": "git+https://github.com/QwantResearch/qwant-basic-gl-style.git#45e60be",
+    "@qwant/qwant-basic-gl-style": "git+https://github.com/QwantResearch/qwant-basic-gl-style.git#04752d4",
```
Comment (amatissart) [inline package.json]: this change should not be part of this PR, I think
